src/rocdecode/roc_decoder_caps.h

```diff
@@ -29,80 +29,214 @@ THE SOFTWARE.
+                va_config_attrib.type = VAConfigAttribRTFormat;
+                CHECK_VAAPI(vaGetConfigAttributes(va_display, profile_list[i], VAEntrypointVLD, &va_config_attrib, 1));
+                CHECK_VAAPI(vaCreateConfig(va_display, profile_list[i], VAEntrypointVLD, &va_config_attrib, 1, &va_config_id));
+                CHECK_VAAPI(vaQuerySurfaceAttributes(va_display, va_config_id, 0, &attr_count));
+                attr_list.resize(attr_count);
+                CHECK_VAAPI(vaQuerySurfaceAttributes(va_display, va_config_id, attr_list.data(), &attr_count));
+                for (int k = 0; k < attr_count; k++) {
+                    switch (attr_list[k].type) {
+                    case VASurfaceAttribPixelFormat:
+                    {
+                        switch (attr_list[k].value.value.i) {
+                            case VA_FOURCC_NV12:
+                                decode_cap_list_[j].output_format_mask |= 1 << rocDecVideoSurfaceFormat_NV12;
+                                break;
+                            case VA_FOURCC_P016:
+                                decode_cap_list_[j].output_format_mask |= 1 << rocDecVideoSurfaceFormat_P016;
+                                break;
+                            default:
+                                break;
+                        }
+                    }
+                        break;
+                    case VASurfaceAttribMinWidth:
+                        if (decode_cap_list_[j].min_width == 0 || (decode_cap_list_[j].min_width > 0 && decode_cap_list_[j].min_width > attr_list[k].value.value.i)) {
+                            decode_cap_list_[j].min_width = attr_list[k].value.value.i;
+                        }
+                        break;
+                    case VASurfaceAttribMinHeight:
+                        if (decode_cap_list_[j].min_height == 0 || (decode_cap_list_[j].min_height > 0 && decode_cap_list_[j].min_height > attr_list[k].value.value.i)) {
+                            decode_cap_list_[j].min_height = attr_list[k].value.value.i;
+                        }
+                        break;
+                    case VASurfaceAttribMaxWidth:
+                        if (decode_cap_list_[j].max_width < attr_list[k].value.value.i) {
+                            decode_cap_list_[j].max_width = attr_list[k].value.value.i;
+                        }
+                        break;
+                    case VASurfaceAttribMaxHeight:
+                        if (decode_cap_list_[j].max_height < attr_list[k].value.value.i) {
+                            decode_cap_list_[j].max_height = attr_list[k].value.value.i;
+                        }
+                        break;
+                    default:
+                        break;
+                    }
+                }
+            }
+        }
+
+        initialized_ = true;
+        return ROCDEC_SUCCESS;
+    }
+    rocDecStatus GetDecoderCaps(RocdecDecodeCaps *pdc) {
+        if (!initialized_) {
```

Comment:
Using the `initialized` variable to call `ProbeHwDecodeCapabilities()` only once is not correct, as it disrupts the Multi-GPU support when multiple GPUs with different capabilities are present in a system. As previously mentioned, you can use the `pdc->device_id` to construct the `drm_node` within the `ProbeHwDecodeCapabilities()` function. If consecutive calls of the `GetDecoderCaps` use the same `pdc->device_id`, you can skip calling the `ProbeHwDecodeCapabilities()`. On the other hand, if the consecutive calls use different `pdc->device_id`, then it's necessary to call `ProbeHwDecodeCapabilities()` to obtain the correct capability of each device.

Reply:
Yup same discussion point on the device_id. Let's discuss.